### PR TITLE
Adds tests for transaction validator and block validators 

### PR DESCRIPTION
**Pull request by talelbaz**
part of #1314 
Old tests that not merge yet and change them to fit the new design( #1008).
There are two parts:
1. Tests for transaction validator.
2. Tests for blockValidator - there were tests that already been covered in the current version, and those that weren't, I relocated them into existing file in the current version: block_body_in_isolation_test. I put the old file test all in comments.

**There is a test that I not finished yet and ill open a new PR for him (validateDifficulty on ValidatePruningPointViolationAndProofOfWorkAndDifficulty_test).

### CODE REVIEW

File: domain/consensus/processes/transactionvalidator/transactionvalidator_test.go [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+				tx:                       &validTx,
+				povBlockHash:             model.VirtualBlockHash,
+				selectedParentMedianTime: 1,
+				isValid:                  true,
+				expectedError:            nil,
+			},
+			{
+				name:                     "checkTransactionCoinbaseMaturity",
+				tx:                       &txForCoinbaseMaturityCheck,
+				povBlockHash:             povBlockHash,
```
Comment (someone235) [inline domain/consensus/processes/transactionvalidator/transactionvalidator_test.go:200]: Why do you use this hash?

File: domain/consensus/processes/transactionvalidator/transactionvalidator_test.go [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+		}
+
+		validTx := externalapi.DomainTransaction{
+			Version:      constants.MaxTransactionVersion,
+			Inputs:       []*externalapi.DomainTransactionInput{&txInputWithMaxSequence},
+			Outputs:      []*externalapi.DomainTransactionOutput{&txOut},
+			SubnetworkID: subnetworks.SubnetworkIDRegistry,
+			Gas:          0,
+			LockTime:     0}
+		txForCoinbaseMaturityCheck := externalapi.DomainTransaction{
```
Comment (someone235) [inline domain/consensus/processes/transactionvalidator/transactionvalidator_test.go:120]: Can you rename this to explain what's special about the transaction, and not for what test it's used? e.g `txWithImmatureCoinbase`. And you should also explain why it's immature (i.e that the UTXO blue score is 5, and the virtual blue score is 1, or something like this)

File: domain/consensus/processes/transactionvalidator/transactionvalidator_test.go [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+			Version:      constants.MaxTransactionVersion,
+			Inputs:       []*externalapi.DomainTransactionInput{&txInput},
+			Outputs:      []*externalapi.DomainTransactionOutput{&txOut},
+			SubnetworkID: subnetworks.SubnetworkIDRegistry,
+			Gas:          0,
+			LockTime:     0}
+		txForInputsAmountsCheck := externalapi.DomainTransaction{
```
Comment (someone235) [inline domain/consensus/processes/transactionvalidator/transactionvalidator_test.go:127]: Can you rename this to explain what's special about the transaction, and not for what test it's used?

File: domain/consensus/processes/transactionvalidator/transactionvalidator_test.go [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+			Version:      constants.MaxTransactionVersion,
+			Inputs:       []*externalapi.DomainTransactionInput{&txInput, &txInputWithLargeEntry},
+			Outputs:      []*externalapi.DomainTransactionOutput{&txOut},
+			SubnetworkID: subnetworks.SubnetworkIDRegistry,
+			Gas:          0,
+			LockTime:     0}
+		txForOutputsAmountsCheck := externalapi.DomainTransaction{
```
Comment (someone235) [inline domain/consensus/processes/transactionvalidator/transactionvalidator_test.go:134]: Can you rename this to explain what's special about the transaction, and not for what test it's used?

File: domain/consensus/processes/transactionvalidator/transactionvalidator_test.go [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+			Version:      constants.MaxTransactionVersion,
+			Inputs:       []*externalapi.DomainTransactionInput{&txInput},
+			Outputs:      []*externalapi.DomainTransactionOutput{&txOutBigValue},
+			SubnetworkID: subnetworks.SubnetworkIDRegistry,
+			Gas:          0,
+			LockTime:     0}
+		txForSequenceLockCheck := externalapi.DomainTransaction{
```
Comment (someone235) [inline domain/consensus/processes/transactionvalidator/transactionvalidator_test.go:141]: Can you rename this to explain what's special about the transaction, and not for what test it's used?

File: domain/consensus/processes/transactionvalidator/transactionvalidator_test.go [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+			Version:      constants.MaxTransactionVersion,
+			Inputs:       []*externalapi.DomainTransactionInput{&txInput},
+			Outputs:      []*externalapi.DomainTransactionOutput{&txOut},
+			SubnetworkID: subnetworks.SubnetworkIDRegistry,
+			Gas:          0,
+			LockTime:     0}
+		txForScriptsCheck := externalapi.DomainTransaction{
```
Comment (someone235) [inline domain/consensus/processes/transactionvalidator/transactionvalidator_test.go:148]: Can you rename this to explain what's special about the transaction, and not for what test it's used?

**Review by someone235 — CHANGES_REQUESTED** [review]

File: domain/consensus/processes/transactionvalidator/transactionvalidator_test.go (hunk 1 above)
Comment (talelbaz) [reply domain/consensus/processes/transactionvalidator/transactionvalidator_test.go:200]: This is an argument of the function ValidateTransactionInContextAndPopulateMassAndFee.

File: domain/consensus/processes/transactionvalidator/transactionvalidator_test.go [hunk 7]
```diff
@@ -117,35 +117,28 @@ func TestValidateTransactionInContextAndPopulateMassAndFee(t *testing.T) {
...
 			LockTime:     0}
-		txForSequenceLockCheck := externalapi.DomainTransaction{
-			Version:      constants.MaxTransactionVersion,
-			Inputs:       []*externalapi.DomainTransactionInput{&txInput},
-			Outputs:      []*externalapi.DomainTransactionOutput{&txOut},
-			SubnetworkID: subnetworks.SubnetworkIDRegistry,
-			Gas:          0,
-			LockTime:     0}
-		txForScriptsCheck := externalapi.DomainTransaction{
+		txWithAnEmptyInvalidScript := externalapi.DomainTransaction{
```
Comment (someone235) [inline domain/consensus/processes/transactionvalidator/transactionvalidator_test.go:141]: I would just call it txWithInvalidSignature

File: domain/consensus/processes/blockvalidator/blockvalidator_test.go [hunk 8]
```diff
@@ -0,0 +1,570 @@
+package blockvalidator
```
[the remark this answers is not in the document]
Comment (someone235) [reply domain/consensus/processes/blockvalidator/blockvalidator_test.go:1]: This is not removed yet

**Review by someone235 — APPROVED** [review]

**Pull request merged**